### PR TITLE
MetaTensor for `GridPatch` and `RandGridPatch`

### LINKED ISSUE #4955: MetaTensor support for `GridPatch` and `RandGridPatch`transform

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #4955 

### Description

This PR makes `GridPatch`, `GridPatchd`, `RandGridPatch` and `RandGridPatchd` to create outputs as MetaTensors and return them as a single MetaTensor with `is_batch=True`.

It also makes `get_default_affine()` a static method, similar to `get_default_meta()`, to be able to it in other places.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/spatial/array.py [hunk 1]
```diff
@@ -3142,6 +3142,9 @@ class GridPatch(Transform):
         pad_mode: refer to NumpyPadMode and PytorchPadMode. If None, no padding will be applied. Defaults to ``"constant"``.
         pad_kwargs: other arguments for the `np.pad` or `torch.pad` function.
 
+    Returns:
+        MetaTensor: A MetaTensor consisting of a batch of all the patches with associated metadata
```
Comment (KumoLiu) [inline monai/transforms/spatial/array.py:3146]: Hi @drbeh, "the associated metadata" just mean the meta data from original data, not pointing to the corresponding patches, right?
```
from monai.transforms import LoadImage, GridPatch, EnsureChannelFirst

image_arr = LoadImage(image_only=True)(image_path)
image1 = EnsureChannelFirst()(image_arr)
image2 = GridPatch(patch_size=(32,32,3), num_patches=10)(image1)

print(image1.meta['spatial_shape'] is image2.meta['spatial_shape']) # True
```
Does `spatial_shape` need to be updated?

File: monai/transforms/spatial/array.py [hunk 2]
```diff
@@ -3265,6 +3270,9 @@ class RandGridPatch(GridPatch, RandomizableTransform):
...
```
Comment (KumoLiu) [inline monai/transforms/spatial/array.py:3274]: The same comments with above.

File: monai/transforms/spatial/array.py (hunk 1 above)
Comment (drbeh) [reply monai/transforms/spatial/array.py:3146]: Good catch, thanks!

File: monai/transforms/spatial/array.py (hunk 2 above)
Comment (drbeh) [reply monai/transforms/spatial/array.py:3274]: Hi @KumoLiu, what do you mean here?

Comment (KumoLiu) [reply monai/transforms/spatial/array.py:3274]: Hi @drbeh, sorry for my misunderstood, just ignore this comment.

File: monai/transforms/spatial/array.py [hunk 3]
```diff
@@ -3225,21 +3228,24 @@ def __call__(self, array: NdarrayOrTensor):
...
+                )
+                locations = np.pad(locations, [[0, padding], [0, 0]], constant_values=0)
+
+        # Convert to MetaTensor
+        metadata = array.meta if isinstance(array, MetaTensor) else MetaTensor.get_default_meta()
+        metadata[WSIPatchKeys.LOCATION] = locations.T
+        metadata[WSIPatchKeys.COUNT] = len(locations)
+        metadata["spatial_shape"] = np.tile(np.array(self.patch_size), (len(locations), 1)).T
+        output = MetaTensor(x=patched_image, meta=metadata)
+        output.is_batch = True
```
Comment (Nic-Ma) [inline monai/transforms/spatial/array.py:3248]: Hi @wyli ,

Could you please help double confirm whether it's `is_batch=True`?

Thanks in advance.

**Review by Nic-Ma — APPROVED** [review]

File: tests/test_grid_patch.py [hunk 4]
```diff
@@ -46,34 +47,69 @@
...
+    TEST_CASES.append([p, *TEST_CASE_4])
+    TEST_CASES.append([p, *TEST_CASE_5])
+    TEST_CASES.append([p, *TEST_CASE_6])
+    TEST_CASES.append([p, *TEST_CASE_7])
+    TEST_CASES.append([p, *TEST_CASE_8])
+    TEST_CASES.append([p, *TEST_CASE_9])
+    TEST_CASES.append([p, *TEST_CASE_10])
+    TEST_CASES.append([p, *TEST_CASE_11])
+    TEST_CASES.append([p, *TEST_CASE_12])
+    TEST_CASES.append([p, *TEST_CASE_13])
```
Comment (wyli) [inline tests/test_grid_patch.py:84]: these changes are not very meaningful to me, any particular reason to revise these?

**Pull request merged**